src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -180,7 +201,8 @@ size_t runtime::gpu::CUDNNEmitter::build_reduce_forward(const cudnnReduceTensorO
     }
 
     auto& desc = m_descriptors.build<cudnnReduceTensorDescriptor_t>();
-    cudnnDataType_t data_type = get_cudnn_datatype(input_type);
+    auto modified_input_type = (input_type == element::i32) ? element::f64 : input_type;
```

Comment:
why convert i32 to f64?

Reply:
Since f32 would not be able to capture the complete range of i32 (only upto 2^24), we need to use f64. Hence, when the input is i32, we convert it to f64, and create descriptors accordingly